product_contract/models/sale_order_line.py

```diff
@@ -12,3 +13,32 @@ class SaleOrderLine(models.Model):
         comodel_name='account.analytic.account',
         string='Contract'
     )
+
+    @api.multi
+    def create_contract(self):
+        """Create contract for sale order line.
+
+        Should be called on confirmation of sale order.
+        """
+        for this in self:
```

Comment:
you can add a filtered in order to avoid the if in the loop, also, its better if we avoid random names, can you change **this** for **line** or something that makes a reference to the singularity of the object?

```python
for line in self.filtered(lambda r: not r.product_id.is_contract:
```

Reply:
Done, but simplified and corrected filter.